community/random-values/src/main/java/org/neo4j/values/storable/RandomValues.java

```diff
@@ -0,0 +1,1310 @@
+    {
+        // Based on Java period (years, months and days)
+        return duration( Period.of( generator.nextInt(), generator.nextInt( 12 ), generator.nextInt( 28 ) ) );
+    }
+
+    /**
+     * Returns the next pseudorandom {@link DurationValue} based on duration.
+     *
+     * @return the next pseudorandom {@link DurationValue}.
+     */
+    public DurationValue nextDuration()
+    {
+        // Based on java duration (seconds)
+        return duration( Duration.of( nextLong( DAYS.getDuration().getSeconds() ), ChronoUnit.SECONDS ) );
+    }
+
+    public Value nextTemporalValue()
+    {
+        int nextInt = generator.nextInt( 6 );
+        switch ( nextInt )
+        {
+        case 0:
+            return nextDateValue();
+
+        case 1:
+            return nextLocalDateTimeValue();
+
+        case 2:
+            return nextDateTimeValue();
+
+        case 3:
+            return nextLocalTimeValue();
+
+        case 4:
+            return nextTimeValue();
+
+        case 5:
+            return nextDuration();
+
+        default:
+            throw new IllegalArgumentException( nextInt + " not a valid temporal type" );
+        }
+    }
+
+    /**
+     * Returns the next pseudorandom two-dimensional cartesian {@link PointValue}.
+     *
+     * @return the next pseudorandom two-dimensional cartesian {@link PointValue}.
+     */
+    public PointValue nextCartesianPoint()
+    {
+        return Values.pointValue( CoordinateReferenceSystem.Cartesian, generator.nextDouble(), generator.nextDouble() );
+    }
+
+    /**
+     * Returns the next pseudorandom three-dimensional cartesian {@link PointValue}.
+     *
+     * @return the next pseudorandom three-dimensional cartesian {@link PointValue}.
+     */
+    public PointValue nextCartesian3DPoint()
+    {
+        return Values.pointValue( CoordinateReferenceSystem.Cartesian_3D, generator.nextDouble(),
+                generator.nextDouble(), generator.nextDouble() );
+    }
+
+    /**
+     * Returns the next pseudorandom two-dimensional geographic {@link PointValue}.
+     *
+     * @return the next pseudorandom two-dimensional geographic {@link PointValue}.
+     */
+    public PointValue nextGeographicPoint()
+    {
+        double longitude = generator.nextDouble() * 360.0 - 180.0;
+        double latitude = generator.nextDouble() * 180.0 - 90.0;
+        return Values.pointValue( CoordinateReferenceSystem.WGS84, longitude, latitude );
+    }
+
+    /**
+     * Returns the next pseudorandom three-dimensional geographic {@link PointValue}.
+     *
+     * @return the next pseudorandom three-dimensional geographic {@link PointValue}.
+     */
+    public PointValue nextGeographic3DPoint()
+    {
+        double longitude = generator.nextDouble() * 360.0 - 180.0;
+        double latitude = generator.nextDouble() * 180.0 - 90.0;
+        return Values.pointValue( CoordinateReferenceSystem.WGS84_3D, longitude, latitude,
+                generator.nextDouble() * 10000 );
```

Comment:
@craigtaverner @sherfert I can't see any limitations or assertions in the production code similar to these... would it be an error to produce a `longitude > 180` or a `height < 0`? The only requirement I see in production is that the values are finite.

Reply:
We don't enforce such a constraint on creation. Everything should work fine, except that the distance function will return nonsense values.

Reply:
But negative height... that might even be expected for submarine positions of something? Excavation sites? Mines? Maybe we should explicitly create those values here then.